notebooks/MAST/HSC/CASJOBS/HSCV3_SMC_API/hscv3_smc_api.ipynb

```diff
@@ -0,0 +1,516 @@
+    "\n",
+    "`select_subset()` selects a sample of points in crowded regions to speed up plotting. This converts the fastkde PDF values (which are probabilities per unit area using the x and y data units) to the expected number of points within a region covered by the marker on the plot.  It then picks only a few points to plot in crowded regions while selecting all the points in uncrowded regions.\n",
+    "\n",
+    "Inputs: \n",
+    "* `zs` = array of PDF density values computed using fastkde\n",
+    "* `axes` = axis range tuple from fastkde\n",
+    "* `markersize` = matplotlib marker size (which is an area in points)\n",
+    "* `over` = oversampling factor (empirically 5 fills in the holes)\n",
+    "* `verbose` = print additional info\n",
+    "\n",
+    "Returns an array of subscripts to plot from `zs`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def select_subset(zs,axes,markersize=2,over=5,verbose=False):\n",
+    "    \"\"\"Select a subset of points in crowded regions to speed up plotting\n",
+    "    \n",
+    "    Returns array of subscripts to plot from zs    \n",
+    "    \"\"\"\n",
+    "    # get figure size in points\n",
+    "    fig = plt.gcf()\n",
+    "    figsize = fig.get_size_inches()*72.0\n",
+    "    # scale factor to convert PDF values to expected number of points in the marker area\n",
+    "    # divide ranges by 2 because fastkde pads the data ranges to allow FFTs\n",
+    "    range0 = 0.5*(axes[0][-1]-axes[0][0])\n",
+    "    range1 = 0.5*(axes[1][-1]-axes[1][0])\n",
+    "    ss = len(zs) * range0 * range1 * markersize / (over*figsize[0]*figsize[1])\n",
+    "    cc = (1.0/(ss*zs).clip(min=1)).cumsum()\n",
+    "    wsel = np.searchsorted(cc, np.arange(np.ceil(cc[-1]).astype(int)+1),\n",
+    "                           side='right').clip(max=len(zs)-1)\n",
+    "    if verbose:\n",
+    "        print(f\"Plotting {len(wsel)} of {len(zs)} points\")\n",
+    "    return wsel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Find objects in the SMC <a name=\"smc\"></a>\n",
+    "\n",
+    "This is based on [HSC Use Case #2](https://archive.stsci.edu/hst/hsc/help/use_case_2_v3.html), which includes an example of creating a color-magnitude diagram for the SMC using <a href=\"https://mastweb.stsci.edu/hcasjobs\">MAST CasJobs</a>.  This is simple to do using the HSC API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Use `astropy` name resolver to get position of the SMC <a name=\"resolver\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "target = 'SMC'\n",
+    "coord = SkyCoord.from_name(target)\n",
+    "\n",
+    "ra = coord.ra.degree\n",
+    "dec = coord.dec.degree\n",
+    "print(f'ra: {ra}\\ndec: {dec}')"
```

Comment:
```suggestion
    "coord\n"
```
Same comment as the last one - use `coord` directly to simplify the number of variables that need to be carried around